alpenhorn/acquisition.py

```diff
@@ -25,6 +32,114 @@ class AcqType(base_model):
     name = pw.CharField(max_length=64)
     notes = pw.TextField(null=True)
 
+    # This dict is an atribute on thr class, used to hold the set of registered
+    # handlers. Store as a dictionary for easy lookup of handlers by name.
+    _registered_acq_types = {}
+
+    @classmethod
+    def register_type(cls, acq_info):
+        """Register a new acquisition type with alpenhorn.
+
+        This creates the entry in the AcqType table if it does not already exist.
+
+        Parameters
+        ----------
+        acq_info : AcqInfoBase
+            AcqInfoBase describing the type of acquisition.
+        """
+
+        try:
+            cls.get(name=acq_info._acq_type)
+        except pw.DoesNotExist:
+            log.info("Create AcqType entry for \"%s\"" % acq_info._acq_type)
+            cls.create(name=acq_info._acq_type)
+
+        # Add to registry
+        cls._registered_acq_types[acq_info._acq_type] = acq_info
+
+    @classmethod
+    def check_registration(cls):
+        """Check that all AcqTypes known to the database have a
+        registered handler.
+        """
+
+        # Get the list of types names from the database
+        db_acq_types = [row[0] for row in cls.select(cls.name).tuples()]
+
+        # Get the names of all the registered types
+        reg_acq_types = cls._registered_acq_types.keys()
+
+        # Find any missing types
+        missing_acq_types = set(db_acq_types) - set(reg_acq_types)
+
+        if len(missing_acq_types):
+            raise RuntimeError('AcqTypes %s have no registered handler.' %
+                               repr(missing_acq_types))
+
+    def is_type(self, acqname, node):
```

Comment:
This method doesn't do much, why not do the check in `detect` where it calls this method?